configure.ac

```diff
@@ -965,6 +965,11 @@ if test "$use_hardening" != "no"; then
       ;;
   esac
 
+  case $host in
+    *aarch64-apple*)
+      AX_CHECK_COMPILE_FLAG([-mbranch-protection=bti], [HARDENED_CXXFLAGS="$HARDENED_CXXFLAGS -mbranch-protection=bti"])
```

Comment:
Why constrain this to apple?

Can we move it up above and just let it fail when unsupported? According to the link above "That instruction is a no-op on systems without BTI".

Reply:
Yea, I think we can just add this. Unlike #24123 (which also does pointer auth), we shouldn't have to worry about additional build issues/warnings here, and can just let configure decide wether to try and use it or not. Although, I'm not sure we'll be able to test for this (in release builds) yet, without additional changes. I'll simplify the changes here now in any case.

Reply:
> Why constrain this to apple?

Not to apple, but I guess one reason to scope this (to aarch64) would be to avoid `Wunused-command-line-argument` warnings on x86_64. Which now appear in the CI (https://github.com/bitcoin/bitcoin/actions/runs/6381638188/job/17318511090?pr=28459#step:7:1979):

```bash
   CXX      libbitcoin_node_a-deploymentstatus.o
clang: error: argument unused during compilation: '-mbranch-protection=bti' [-Werror,-Wunused-command-line-argument]
```

Reply:
Scoped this back to aarch64.

Reply:
Blah, agreed.